activemq-client/src/test/java/zipkin2/reporter/activemq/ActiveMQExtension.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2016-2023 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.reporter.activemq;
+
+import java.time.Duration;
+import org.apache.activemq.ActiveMQConnectionFactory;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.opentest4j.TestAbortedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import static org.testcontainers.utility.DockerImageName.parse;
+
+class ActiveMQExtension implements BeforeAllCallback, AfterAllCallback {
+  static final Logger LOGGER = LoggerFactory.getLogger(ActiveMQExtension.class);
+  static final int ACTIVEMQ_PORT = 61616;
+
+  ActiveMQContainer container = new ActiveMQContainer();
+
+  @Override public void beforeAll(ExtensionContext context) {
+    if (context.getRequiredTestClass().getEnclosingClass() != null) {
+      // Only run once in outermost scope.
+      return;
+    }
+
+    container.start();
+    LOGGER.info("Using brokerURL " + brokerURL());
+  }
+
+  @Override public void afterAll(ExtensionContext context) {
+    if (context.getRequiredTestClass().getEnclosingClass() != null) {
+      // Only run once in outermost scope.
+      return;
+    }
+
+    container.stop();
+  }
+
+  ActiveMQSender.Builder newSenderBuilder(String queue) {
+    ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory();
+    connectionFactory.setBrokerURL(brokerURL());
+    return ActiveMQSender.newBuilder().queue(queue).connectionFactory(connectionFactory);
+  }
+
+  String brokerURL() {
+    return "failover:tcp://" + container.getHost() + ":" + container.getMappedPort(ACTIVEMQ_PORT);
+  }
+
+  // mostly waiting for https://github.com/testcontainers/testcontainers-java/issues/3537
```

Comment:
I think this may be copy-pasted and no need to change anything now, but just for reference, `@TestContainers` and `@Container` automatically skip when docker isn't available. It would look something like this, but don't think it's a priority to change anything

Rename `ActiveMQExtension` -> `ActiveMQContainer`

```java
public static class ActiveMQContainer extends ... {
  public ActiveMQContainer() {
    ...
  }

  public String newSenderBuilder(String queue) {
    ...
  }
}

@Testcontainers
public class ITActiveMQSender {
  @Container public static final activemq = new ActiveMQContainer();

...
}
```

Reply:
does this also allow property override? in the main zipkin build we intentionally skip docker on one pass so we can parallelize them...

Reply:
I think if we could do what you said and still have a skip property (even if we make it), sounds much better and I can help sweep around and apply the pattern everywhere.

Reply:
I'll try a separate PR once this is in to do all of them. thanks for the tip @anuraaga!